product_multi_barcode_constraint_per_company/models/product_barcode.py

```diff
@@ -0,0 +1,21 @@
+# Copyright 2023 Cetmix OÜ
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import api, models
+
+
+class ProductBarcode(models.Model):
+    _inherit = "product.barcode"
+
+    @api.constrains("name")
+    def _check_duplicates(self):
+        for record in self:
+            barcodes = self.search(
```

Comment:
Why you use search function here?
And why you call `super` in cycle?

Reply:
Thanks, I'll fix it